api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -513,14 +532,18 @@ type HostedClusterSpec struct {
 
 	// ImageContentSources specifies image mirrors that can be used by cluster
 	// nodes to pull content.
-	//
+	// When imageContentSources is set, the controllers will generate a machineConfig.
+	// This MachineConfig will be part of every payload generated by the controllers for any NodePool of the HostedCluster.
+	// Changing this value will trigger a rollout for all existing NodePools in the cluster.
 	// +optional
-	// +immutable
 	ImageContentSources []ImageContentSource `json:"imageContentSources,omitempty"`
 
-	// AdditionalTrustBundle is a reference to a ConfigMap containing a
-	// PEM-encoded X.509 certificate bundle that will be added to the hosted controlplane and nodes
-	//
+	// AdditionalTrustBundle is a local reference to a ConfigMap that must have a "ca-bundle.crt" key
+	// whose content must be a PEM-encoded X.509 certificate bundle that will be added to the hosted controlplane and nodes
+	// If the reference is set but none of the above requirements are met, the HostedCluster will enter a degraded state.
+	// TODO(alberto): Signal this in a condition.
```

Comment:
nit: would be nice to link a Jira ticket here so we don't lose track of this TODO.

Reply:
This will be a different activity